core/src/com/po/fuck/model/weapons/LaserGun.java

```diff
@@ -2,20 +2,23 @@
 
 import static com.po.fuck.model.Constants.LASERGUN_COOLDOWN;
 
-import com.badlogic.gdx.graphics.Texture;
-import com.badlogic.gdx.graphics.g2d.Sprite;
 import com.badlogic.gdx.math.Vector2;
+import com.po.fuck.Assets;
+import com.po.fuck.model.loadersAssets.WeaponSpriteInfo;
 import com.po.fuck.model.Entity;
 
 public final class LaserGun extends Gun {
     {
         cooldown = LASERGUN_COOLDOWN;
-        geometryData.setHeight(new Sprite(new Texture("laser_gun2.png")).getHeight());
-        geometryData.setWidth(new Sprite(new Texture("laser_gun2.png")).getWidth());
     }
 
     public LaserGun(Entity owner) {
         super(owner);
+        name = "laserGun";
+
+        WeaponSpriteInfo info = Assets.getWeaponAssetInfo(name);
+        geometryData.setHeight(info.height);
+        geometryData.setWidth(info.width);
```

Comment:
What is the idea behind moving it to constructor?

Reply:
well it will be in initialiser